docs/ocis/adr/0021-service-authorization.md

```diff
@@ -0,0 +1,127 @@
+Maybe not, AFAICT we could internally use spicedb and use it to implement the permission checks.
+
+#### Why can't we just mint a token and bake the permissions the service has into it.
+For normal users this is called the scope and it covers the permissions the use has consented to. When an access token arrives at the proxy we can use the scope claim which is filled by the IdP to limit the request to that scope. However, we will not be able to teach every IdP every scope that users should be able to consent to. It is far more practical to read a role claim and mint the scopes based on that, e.g. a role user would get the `Files.ReadWrite.All` permission that scopes the request to read all his files. A service that uses delegated permission can during this request only access the users files and all files the user has access to. We can add a scope middleware to the graph api (or other http endpoints) that checks if the user has consonted to the scope required for the request.
+
+This does not free us from the actual permission check to determine if the user can actually read or write the file. While `Files.ReadWrite.All` means that the service could write to a file that was shared with the logged in user an actual permission check needs to determine if the file was shared with write permission or if it was read only.
+
+#### How do we check the service account has the permission to make a call?
+In the same way we currently check permissions. We delegate that to the permissions service. Hm but that means we need to make two permission checks: one for the service account and one for the logged in user, if present. We need to determine if the service is allowed to make a request and if the user is allowed to make a request. We want to protect against .g. a compromised thumbnail service that should only make read requests but then tries to write a file. This needs to be checked at the storage provider.
+
+What if we logically AND the delegated permission with the users scope?
+
+How does go micro solve this?
+
+To be clear: this has nothing to do with the scope.
+
+
+#### How do we "provision" service accounts?
+
+#### How/where do we assign roles/permissions to them?
+
+ 
+How do services declare what permissions they need?
+  Especially useful when the admin wants to use a (maybe closed source) 3rd party extension. (Some day in the future)
+
+* Good, because we could replace machine auth with specific service accounts and no longer have to distribude a shared secret everywhere
+* Bad, because we don't know if a there are places in the code that try to look up a user with USER_TYPE_SERVICE at the cs3 users service ... they might not exist there ... or do we have to implement a userregistry, similar to the authregistry?
+* Bad, because we have to provision and manage service accounts on init
+* Bad, because we have to write codemanage service accounts in the admin ui
+
+
+### Auth Manager type Space-Owner
```

Comment:
This heading does not make sense for me, maybe my fault, but still...

Reply:
Changed to `Impersonate Space-Owners`